src/libs/PersonalDetailsUtils.ts

```diff
@@ -77,7 +84,7 @@ function getPersonalDetailsByIDs(accountIDs: number[], currentUserAccountID: num
 }
 
 function getPersonalDetailByEmail(email: string): PersonalDetails | undefined {
-    return (Object.values(allPersonalDetails ?? {}) as PersonalDetails[]).find((detail) => detail?.login === email);
+    return emailToPersonalDetailsCache[email.toLowerCase()];
```

Comment:
Let us have a test case around verifying `policyApproverName` [here](https://github.com/Expensify/App/blob/ebc8de8229fa700e5a7d08f4589db934a79d4f1d/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx#L52C67-L52C91) in Workflows page which uses `getPersonalDetailByEmail`

Reply:
Okay added some test steps to test [this](https://github.com/Expensify/App/blob/ebc8de8229fa700e5a7d08f4589db934a79d4f1d/src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx#L52)